tests/integ/test_session.py

```diff
@@ -344,3 +344,20 @@ def check(schema_name: str, expected_name: str) -> None:
     check("A", '"A"')
     check('"a b"', '"a b"')
     check('"a""b"', '"a""b"')
+
+
+def test_use_secondary_roles(session):
+    session.use_secondary_roles("all")
+    session.use_secondary_roles("none")
+    session.use_secondary_roles("ALL")
+    session.use_secondary_roles("NONE")
+
+    current_role = session.get_current_role()
+    with pytest.raises(
+        ProgrammingError,
```

Comment:
Do we still throw ProgrammingError to users?

Reply:
Yea, only methods decorated by `@SnowflakePlan.Decorator.wrap_exception` will wrap ProgrammingError to SnowparkSQLException. Here we use run_query in server_connection.py, which is not decorated by `@SnowflakePlan.Decorator.wrap_exception`.

Reply:
In theory it is. Little to no impact to users though.